File: core/src/main/java/pcb/uwu/units/derived/termodynamics/PressureUnit.java
package pcb.uwu.units.derived.termodynamics;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.UnitCounter;
import pcb.uwu.units.base.LengthUnit;
import pcb.uwu.units.base.MassUnit;
import pcb.uwu.units.base.TimeUnit;
import pcb.uwu.units.derived.fundamental.AreaUnit;
import pcb.uwu.units.derived.mechanics.ForceUnit;

public class PressureUnit extends CompositeUnit {

	// region base constructor

	public PressureUnit(LengthUnit lengthUnit, MassUnit massUnit, TimeUnit timeUnit) {
		super(new UnitCounter()
				.major(massUnit)
				.minor(lengthUnit)
				.minor(timeUnit, 2));
	}

	// endregion

	// region derived constructors

	public PressureUnit(ForceUnit forceUnit, AreaUnit areaUnit) {
		super(new UnitCounter()
				.major(forceUnit.getUnitCounter())
				.minor(areaUnit.getUnitCounter()));
	}

	// endregion

}

File: dimension/src/main/java/pcb/uwu/unit/derived/termodynamics/PressureUnit.kt
package pcb.uwu.unit.derived.termodynamics

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.LengthUnit
import pcb.uwu.unit.base.MassUnit
import pcb.uwu.unit.base.TimeUnit
import pcb.uwu.unit.derived.geometry.AreaUnit
import pcb.uwu.unit.derived.geometry.VolumeUnit
import pcb.uwu.unit.derived.mechanics.ForceUnit

open class PressureUnit : CompositeUnit
{
    constructor(lengthUnit: LengthUnit,
                massUnit: MassUnit,
                timeUnit: TimeUnit)
            : super(UnitCounter()
                        .major(massUnit)
                        .minor(lengthUnit)
                        .minor(timeUnit)
                        .minor(timeUnit))

    constructor(forceUnit: ForceUnit,
                areaUnit: AreaUnit)
            : super(UnitCounter()
                        .major(forceUnit)
                        .minor(areaUnit))

    constructor(energyUnit: EnergyUnit,
                volumeUnit: VolumeUnit)
            : super(UnitCounter()
                        .major(energyUnit)
                        .minor(volumeUnit))
}
